lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HadoopPaths.java

```diff
@@ -31,12 +30,22 @@ public static Path hadoopPath(Location location)
         String path = location.toString();
         Path hadoopPath = new Path(path);
         if ("s3".equals(hadoopPath.toUri().getScheme()) && !path.equals(hadoopPath.toString())) {
-            if (hadoopPath.toUri().getFragment() != null) {
-                throw new IllegalArgumentException("Unexpected URI fragment in path: " + path);
-            }
-            URI uri = URI.create(path);
-            return new Path(uri + "#" + URLEncoder.encode(uri.getPath(), UTF_8));
+            return new Path(toPathEncodedUri(location));
         }
         return hadoopPath;
     }
+
+    private static URI toPathEncodedUri(Location location)
+    {
+        try {
+            return new URI(
+                    location.scheme().orElse(null),
+                    location.host().orElse(null),
+                    "/" + location.path(),
```

Comment:
why prefix with `/`? add a code comment

Reply:
`Location.path()` is documented to not start with a slash